driver/xrt/include/constants.hpp

```diff
@@ -28,6 +28,9 @@ namespace ACCL {
 typedef uint64_t addr_t;
 /** Device value type */
 typedef uint32_t val_t;
+/** Communicator IDs used to refer to a certain communicator */
+typedef unsigned int CommunicatorId;
```

Comment:
I would suggest renaming this to `communicatorId` to keep it more consistent with the rest of ACCL (i.e. `dataType` and `reduceFunction`.